### PR TITLE
Swap out tox for nox

### PR DESCRIPTION
**Pull request by anselor**
Nox is like tox but implemented in the form of a python file with the full power of python.

### CODE REVIEW

**Comment by tleonhardt** [comment]
@anselor I have never heard of nox before.  Can you please elaborate on what you think the advantages are in comparison to tox?  Are you aware of any down sides?

@kotfu Do you have any experience with nox?

**Comment by anselor** [comment]
It's inspired by tox but is fully python. Allows you to have control over the generation of venvs and run commands inside them. This would allow control over which specific wheels are installed in each venv and allow tests to be written to validate behavior with and without plug-ins.

**Comment by tleonhardt** [comment]
Ok, that sounds pretty decent.  Are there any known disadvantages relative to tox?

**Comment by tleonhardt** [comment]
Assuming we do make the switch, there are a few additional files which should be updated.  GitHub search in our repo for tox shows 9 files.  There are additional files like tasks.py, MANIFEST.in, .gitignore, setup.cfg, etc.  we should also delete the tox.ini unless we want to leave it around in-case.

**Comment by anselor** [comment]
My understanding is tox has lots of magic it does that makes the simplest cases very simple but becomes really doesn't let you customize and stray from the path they've defined. 

Nox I think is a slightly higher learning curve to set up, but you really have full access to executing commands as if you're on the console in the venv.

Also, if you look at my other branch I've already bright in the external test plugin and gotten it running under nox.

**Comment by tleonhardt** [comment]
I like that the nox config file is in Python - that seems like a plus.  If you could tie up the loose ends in terms of modifying the remaining files which use tox for equivalent nox usage I’m good to approve.

File: noxfile.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import nox
+
+
+@nox.session(python=['3.7'])
```
Comment (tleonhardt) [inline noxfile.py:4]: Nice work!  Do like the format of this file better.  I find it more readable and less magical for general Python developers who aren't experts in `tox`.

**Pull request merged**